File: data/TiendaMusical/Dependencies/Instrumento.js
/**
 * Clase que representa un instrumento musical.
 * Asume que existe una clase Producto de la que hereda.
 */
class Instrumento extends Producto {
    /**
     * @param {string} id 
     * @param {string} titulo 
     * @param {number} precio 
     * @param {TipoInstrumento} tipo 
     * @param {string} material 
     */
    constructor(id, titulo, precio, tipo, material) {
        super(id, titulo, precio);
        this.tipo = tipo;
        this.material = material;
        /** @type {string[]} */
        this.accesoriosIncluidos = [];
    }

    /**
     * Añade un accesorio incluido con el instrumento.
     * Valida que el accesorio no esté en la lista antes de añadirlo.
     * @param {string} accesorio Nombre del accesorio
     */
    añadirAccesorio(accesorio) {
        if (accesorio && !this.accesoriosIncluidos.includes(accesorio)) {
            this.accesoriosIncluidos.push(accesorio);
        }
    }

    /**
     * Calcula el peso estimado del instrumento.
     * Se asume la siguiente lógica:
     * - Guitarras: 2.5 kg
     * - Bajos: 3.0 kg
     * - Baterías: 10.0 kg
     * - Otros: 5.0 kg
     * @return {number} Peso en kg (estimación)
     */
    estimarPeso() {
        switch (this.tipo) {
            case TipoInstrumento.GUITARRA:
                return 2.5;
            case TipoInstrumento.BAJO:
                return 3.0;
            case TipoInstrumento.BATERIA:
                return 10.0;
            default:
                return 5.0;
        }
    }

    // Getters
    getTipo() {
        return this.tipo;
    }
    
    getMaterial() {
        return this.material;
    }
    
    getAccesoriosIncluidos() {
        return [...this.accesoriosIncluidos]; // Devuelve una copia del array
    }
}

/**
 * Enumeración de tipos de instrumento.
 * En JavaScript podemos simular una enum con un objeto.
 */
const TipoInstrumento = {
    GUITARRA: "GUITARRA",
    BAJO: "BAJO",
    BATERIA: "BATERIA",
    OTRO: "OTRO"
};

File: data/TiendaMusical/Instrumento.java
import java.util.ArrayList;

/**
 * Clase que representa un instrumento musical.
 */
public class Instrumento extends Producto {

    /**
     * Tipo de instrumento (ej. guitarra, bajo, bateria).
     */
    private TipoInstrumento tipo;

    private String material;
    private ArrayList<String> accesoriosIncluidos;

    public Instrumento(String id, String titulo, double precio, TipoInstrumento tipo, String material) {
        super(id, titulo, precio);
        this.tipo = tipo;
        this.material = material;
        this.accesoriosIncluidos = new ArrayList<>();
    }

    /**
     * Añade un accesorio incluido con el instrumento.
     * Debe validar que el accesorio no esté en la lista antes de añadirlo.
     * @param accesorio Nombre del accesorio
     */
    public void añadirAccesorio(String accesorio) {
        //TO DO
    }

    /**
     * Calcula el peso estimado del instrumento.
     * Se asume la siguiente lógica:
     * - Guitarras: 2.5 kg
     * - Bajos: 3.0 kg
     * - Baterías: 10.0 kg
     * - Otros: 5.0 kg
     * @return Peso en kg (estimación)
     */
    public double estimarPeso() {
        // TODO: Implementar estimación según tipo y material
        return 0.0;
    }

    // Setters y Getters
    public TipoInstrumento getTipo() {
        return tipo;
    }
    public String getMaterial() {
        return material;
    }
    public ArrayList<String> getAccesoriosIncluidos() {
        return new ArrayList<>(accesoriosIncluidos);
    }
}